### DISCUSSION TITLE
MAS submission

### DISCUSSION THREAD

**Discussion by zkrige** [nwutils/nw-builder#950] [General]
Has anyone managed to submit an NW.JS app to App Store? I've managed to get notarisation working but when I submit I get 

Asset validation failed (90296)
App sandbox not enabled. The following executables must include the "com.apple.security.app-sandbox" entitlement with a Boolean value of true in the entitlements property list: [( "bundle-id.pkg/Payload/APP.app/Contents/Frameworks/nwjs Framework.framework/Versions/117.0.5938.63/Helpers/app_mode_loader", "bundle-id.pkg/Payload/APP.app/Contents/Frameworks/nwjs Framework.framework/Versions/117.0.5938.63/Helpers/chrome_crashpad_handler", "bundle-id.pkg/Payload/APP.app/Contents/Frameworks/nwjs Framework.framework/Versions/117.0.5938.63/Helpers/nwjs Helper (Alerts).app/Contents/MacOS/nwjs Helper (Alerts)", "bundle-id.pkg/Payload/APP.app/Contents/Frameworks/nwjs Framework.framework/Versions/117.0.5938.63/Helpers/nwjs Helper (GPU).app/Contents/MacOS/nwjs Helper (GPU)", "bundle-id.pkg/Payload/APP.app/Contents/Frameworks/nwjs Framework.framework/Versions/117.0.5938.63/Helpers/nwjs Helper (Plugin).app/Contents/MacOS/nwjs Helper (Plugin)", "bundle-id.pkg/Payload/APP.app/Contents/Frameworks/nwjs Framework.framework/Versions/117.0.5938.63/Helpers/nwjs Helper (Renderer).app/Contents/MacOS/nwjs Helper (Renderer)", "bundle-id.pkg/Payload/APP.app/Contents/Frameworks/nwjs Framework.framework/Versions/117.0.5938.63/Helpers/nwjs Helper.app/Contents/MacOS/nwjs Helper", "bundle-id.pkg/Payload/APP.app/Contents/MacOS/nwjs" )] Refer to App Sandbox page at https://developer.apple.com/documentation/security/app_sandbox for more information on sandboxing your app. (ID: 3abfe98a-3a95-439d-83fe-e5effac4c1b9)

however, im pretty sure that enabling sandboxing will break a lot of my functionality that reads from disk

**Comment by ayushmanchhabra** [comment]
https://github.com/nwjs/nw.js/issues?q=is%3Aissue+is%3Aopen+mas
Does this help in any way?

**Comment by zkrige** [comment]
Not really. I'm able to sign the package file without problems. However MAS needs sandbox turned on. And the moment you do that, it introduces a host of other problems into NW.JS. basically nwjs doesn't like running in sandbox mode.
